cmd/simulator/load/funder.go

```diff
@@ -0,0 +1,106 @@
+// Copyright (C) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package load
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+
+	"github.com/ava-labs/subnet-evm/cmd/simulator/key"
+	"github.com/ava-labs/subnet-evm/core/types"
+	"github.com/ava-labs/subnet-evm/ethclient"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+// DistributeFunds ensures that each address in keys has at least [minFundsPerAddr] by sending funds
+// from the key with the highest starting balance.
+// This function should never return a set of keys with length less than [numKeys]
+func DistributeFunds(ctx context.Context, client ethclient.Client, keys []*key.Key, numKeys int, minFundsPerAddr *big.Int) ([]*key.Key, error) {
+	if len(keys) < numKeys {
+		return nil, fmt.Errorf("insufficient number of keys %d < %d", len(keys), numKeys)
+	}
+	fundedKeys := make([]*key.Key, 0, numKeys)
+	// TODO: clean up fund distribution.
```

Comment:
should we validate that the client is not nil ?

Reply:
no